db-scripts/src/main/resources/migration.sql

```diff
@@ -374,10 +374,106 @@ INSERT INTO mutation_count_by_keyword
 
 UPDATE info SET DB_SCHEMA_VERSION="2.3.1";
 
+
 ##version: 2.4.0
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_FILTER` VARCHAR(20) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_FILTER_ANNOTATION` VARCHAR(80) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_TIERS_FILTER` VARCHAR(50) NULL;
 ALTER TABLE `mutation` ADD COLUMN `DRIVER_TIERS_FILTER_ANNOTATION` VARCHAR(80) NULL;
 
-UPDATE info SET DB_SCHEMA_VERSION="2.4.0";
+-- ========================== new reference genome gene related tables =============================================
+CREATE TABLE `reference_genome` (
+    `reference_genome_id` int(11) NOT NULL AUTO_INCREMENT,
+    `species` varchar(64) DEFAULT NULL,
+    `name` varchar(64) DEFAULT NULL,
+    `build_name` varchar(64) DEFAULT NULL,
+    `genome_size` bigint(20) DEFAULT NULL,
+    `URL` varchar(256) DEFAULT NULL,
+    `release_date` datetime DEFAULT NULL,
+    PRIMARY KEY (`reference_genome_id`),
+    UNIQUE INDEX `NAME_UNIQUE` (`build_name` ASC)
+);
+
+-- ========================== new reference genomes ====================================================================
+INSERT INTO `reference_genome` 
+VALUES (1, 'human', 'hg19', 'GRCh37', NULL, 'http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes', '2009-02-01');
+INSERT INTO `reference_genome` 
+VALUES (2, 'human', 'hg38', 'GRCh38', NULL, 'http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes', '2013-12-01');
+INSERT INTO `reference_genome` 
+VALUES (3, 'mouse', 'mm10', 'GRCm38', NULL, 'http://hgdownload.cse.ucsc.edu//goldenPath/mm10/bigZips/mm10.chrom.sizes', '2012-01-01');
+
+CREATE TABLE `reference_genome_gene` (
+    `entrez_gene_id` int(11) NOT NULL,
+    `cytoband` varchar(64) DEFAULT NULL,
+    `exonic_length` int(11) DEFAULT NULL,
+    `gene_start` bigint(20) DEFAULT NULL,
+    `gene_end` bigint(20) DEFAULT NULL,
+    `gene_stable_ID` varchar(64) DEFAULT NULL,
+    `chr` varchar(64) DEFAULT NULL,
+    `reference_genome_id` int(11) NOT NULL,
+    PRIMARY KEY (`entrez_gene_id`,`reference_genome_id`),
+    FOREIGN KEY (`reference_genome_id`) REFERENCES `reference_genome` (`reference_genome_id`) ON DELETE CASCADE,
+    FOREIGN KEY (`entrez_gene_id`) REFERENCES `gene` (`ENTREZ_GENE_ID`) ON DELETE CASCADE
+);
+
+INSERT INTO reference_genome_gene (entrez_gene_id, cytoband, exonic_length, chr, reference_genome_id)
+(SELECT 
+	ENTREZ_GENE_ID, 
+	CYTOBAND, 
+	LENGTH, 
+	SUBSTRING_INDEX(gene.CYTOBAND,IF(LOCATE('p', gene.CYTOBAND), 'p', 'q'), 1), 
+	1 
+FROM `gene`);
+
+-- ========================== new foreign key references new reference genome table  ===================================
+ALTER TABLE `mutation_event` ADD COLUMN `REFERENCE_GENOME_ID` INT NULL AFTER `NCBI_BUILD`;
+ALTER TABLE `mutation_event` 
+ADD FOREIGN KEY (`REFERENCE_GENOME_ID`) REFERENCES `reference_genome` (`REFERENCE_GENOME_ID`) ON DELETE CASCADE;
```

Comment:
This is specified as ON DELETE RESTRICT in cgds.sql

Reply:
well spotted. Will make a change in cgds.ql to "ON DELETE RESTRICT". Thanks!